exporter/loggingexporter/factory.go

```diff
@@ -43,34 +45,59 @@ func (f *Factory) CreateDefaultConfig() configmodels.Exporter {
 			TypeVal: typeStr,
 			NameVal: typeStr,
 		},
-		LogLevel: "info",
+		LogLevel:           "info",
+		SamplingInitial:    defaultSamplingInitial,
+		SamplingThereafter: defaultSamplingThereafter,
 	}
 }
 
 // CreateTraceExporter creates a trace exporter based on this config.
 func (f *Factory) CreateTraceExporter(logger *zap.Logger, config configmodels.Exporter) (exporter.TraceExporter, error) {
 	cfg := config.(*Config)
 
-	exporterLogger, err := f.createLogger(cfg.LogLevel)
+	exporterLogger, err := f.createLogger(cfg.LogLevel, cfg.SamplingInitial, cfg.SamplingThereafter)
 	if err != nil {
 		return nil, err
 	}
 
-	lexp, err := NewTraceExporter(config, exporterLogger)
+	lexp, err := NewTraceExporter(config, cfg.LogLevel, exporterLogger)
 	if err != nil {
 		return nil, err
 	}
 	return lexp, nil
 }
 
-func (f *Factory) createLogger(logLevel string) (*zap.Logger, error) {
+func (f *Factory) createLogger(logLevel string, initial int, thereafter int) (*zap.Logger, error) {
 	var level zapcore.Level
 	err := (&level).UnmarshalText([]byte(logLevel))
 	if err != nil {
 		return nil, err
 	}
-	conf := zap.NewProductionConfig()
-	conf.Level.SetLevel(level)
+
+	encoderConf := zapcore.EncoderConfig{
+		// Keys can be anything except the empty string.
+		TimeKey:        "T",
+		MessageKey:     "M",
+		LineEnding:     zapcore.DefaultLineEnding,
+		EncodeLevel:    zapcore.CapitalLevelEncoder,
+		EncodeTime:     zapcore.ISO8601TimeEncoder,
+		EncodeDuration: zapcore.StringDurationEncoder,
+		EncodeCaller:   zapcore.ShortCallerEncoder,
+	}
+
+	conf := zap.Config{
+		Level:       zap.NewAtomicLevelAt(level),
+		Development: false,
+		Sampling: &zap.SamplingConfig{
+			Initial:    initial,
+			Thereafter: thereafter,
+		},
+		Encoding:         "console",
+		EncoderConfig:    encoderConf,
+		OutputPaths:      []string{"stderr"},
```

Comment:
If I understand correctly this redirects logs printed that would be previously printed to stdout to stderr instead. Why do we do this?

Reply:
This is based on the default config of `Zap` actually. I replaced that part with taking the config from `Zap` and just updating it